tests/flytekit/unit/cli/pyflyte/test_run.py

```diff
@@ -387,3 +391,34 @@ def test_datetime_type():
     v = t.convert("now", None, None)
     assert v.day == now.day
     assert v.month == now.month
+
+
+def test_json_type():
+    t = JsonParamType()
+    assert t.convert(value='{"a": "b"}', param=None, ctx=None) == {"a": "b"}
+
+    with pytest.raises(click.BadParameter):
+        t.convert(None, None, None)
+
+    # test that it loads a json file
+    with tempfile.NamedTemporaryFile("w") as f:
```

Comment:
`NamedTemporaryFile` are weird on windows. They cannot be opened twice (while the first one is still open - as per the [docs](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile)). A suggested workaround is to set `delete=False` in the calls to instantiate them. 

It's pretty annoying.

Reply:
copilot generated these tests :D - good catch